pandas/__init__.py

```diff
@@ -202,8 +202,29 @@
         FutureWarning,
         stacklevel=2,
     )
-# Don't allow users to use pandas.os or pandas.warnings
-del os, warnings
+# Don't allow users to use pandas.os
+del os
+
+# DeprecationWarning for missing pyarrow
+from pandas.compat.pyarrow import pa_version_under10p1
+
+if pa_version_under10p1:
+    import warnings
```

Comment:
Isn't warnings already imported at the very top of the file?

Reply:
Thanks, updated.